hub/config/src/main/java/uk/gov/ida/hub/config/data/S3ConfigSource.java

```diff
@@ -37,12 +38,14 @@ public S3ConfigSource(ConfigConfiguration configConfiguration, AmazonS3 s3Client
         CacheLoader<String, RemoteConfigCollection> loader;
         loader = new CacheLoader<String, RemoteConfigCollection>() {
             private RemoteConfigCollection currentConfig;
-            private Date lastModified;
 
             @Override
             public RemoteConfigCollection load(String key) {
                 S3Object s3Object = getS3Object(key);
-                if (this.lastModified.before(s3Object.getObjectMetadata().getLastModified())) {
+                if(lastModified == null) {
```

Comment:
If you use `CacheLoader` then you can use [`expireAfterWrite(...)`] and not need to use `lastModified`. Every time the cache expires for that key the next read will force the cache to load a new value.

Your code could look something like this:
```
  LoadingCache<Key, RemoteConfigCollection> graphs = CacheBuilder.newBuilder()
       .maximumSize(1) //optional
       .expireAfterWrite(TIMEOUT, TimeUnit.MINUTES)
       .build(
           new CacheLoader<Key, RemoteConfigCollection>() {
             public Graph load(Key key) throws AnyException {
               return s3client.getConfigFromS3();
             }
           });
```

Reply:
I'm looking at the expireAfterWrite stuff now.

Reply:
Thanks, here's an example from the hub: https://github.com/alphagov/verify-hub/blob/7109a388a896eb8c3aa8fbf303a490b05c8b4b2c/hub/saml-engine/src/main/java/uk/gov/ida/hub/samlengine/config/CertificatesConfigProxy.java#L29